pkg/totalaccountwatcher/totalaccountwatcher.go

```diff
@@ -0,0 +1,155 @@
+func NewTotalAccountWatcher(client client.Client, AwsClient awsclient.Client, watchInterval time.Duration) *totalAccountWatcher {
+	return &totalAccountWatcher{
+		watchInterval: watchInterval,
+		AwsClient:     AwsClient,
+		client:        client,
+	}
+}
+
+// TotalAccountWatcher will trigger AwsLimitUpdate every `scanInternal` and only stop if the operator is killed or a
+// message is sent on the stopCh
+func (s *totalAccountWatcher) Start(log logr.Logger, stopCh <-chan struct{}) {
+	log.Info("Starting the totalAccountWatcher")
+	for {
+		select {
+		case <-time.After(s.watchInterval):
+			log.Info("totalAccountWatcher: updating a")
+			err := s.UpdateTotalAccounts(log)
+			if err != nil {
+				log.Error(err, "totalAccountWatcher not started, awsLimit won't be updated")
+			}
+		case <-stopCh:
+			log.Info("Stopping the totalAccountWatcher")
+			break
+		}
+	}
+}
+
+// AwsLimitUpdate will update the TotalAccountWatcher's total field
+func (s *totalAccountWatcher) UpdateTotalAccounts(log logr.Logger) error {
+
+	// Reset cache
+	secretList := &corev1.SecretList{}
+	secret := &corev1.Secret{}
+
+	listOps := &client1.ListOptions{Namespace: awsv1alpha1.AccountCrNamespace}
+	if err := TotalAccountWatcher.client.List(context.TODO(), listOps, secretList); err != nil {
+		log.Error(err, fmt.Sprintf("Unable to list secrets in namespace %s", awsv1alpha1.AccountCrNamespace))
+	}
+
+	if err := TotalAccountWatcher.client.Get(context.TODO(), types.NamespacedName{Name: AwsSecretName, Namespace: "aws-account-operator"}, secret); err != nil {
+		log.Error(err, fmt.Sprintf("Unable to get secrets in namespace %s", awsv1alpha1.AccountCrNamespace))
+	}
+
+	accountTotal, err := TotalAwsAccounts()
+	if err != nil {
+		log.Error(err, "Failed to get AWS account total from AWS api")
+	}
+
+	TotalAccountWatcher.Total = accountTotal
+	log.Info(fmt.Sprintf("Total AWS accounts: %d", TotalAccountWatcher.Total))
+
+	gciOut, err := TotalAccountWatcher.AwsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	log.Info(fmt.Sprintf("Caller identity: %+v", gciOut))
+
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Failed to get total number of AWS accounts: %s", err))
+	} else {
+		localmetrics.MetricTotalAWSAccounts.With(prometheus.Labels{"name": "aws-account-operator"}).Set(float64(accountTotal))
+	}
+
+	TotalAccountWatcher.Total = accountTotal
+
+	return nil
+}
+
+// TotalAwsAccounts returns the total number of aws accounts in the aws org
+func TotalAwsAccounts() (int, error) {
+	var awsAccounts []*organizations.Account
+
+	var nextToken *string
+
+	// Ensure we paginate through the account list
+	for {
+		awsAccountList, err := TotalAccountWatcher.AwsClient.ListAccounts(&organizations.ListAccountsInput{NextToken: nextToken})
+		if err != nil {
+			errMsg := "Error getting a list of accounts"
+			if aerr, ok := err.(awserr.Error); ok {
+				errMsg = fmt.Sprintf("Failed to get account list with error code %s", aerr.Message())
+			}
+			return 0, errors.New(errMsg)
+		}
```

Comment:
This probably needs a refactor now since we don't ever want to return `0` if we get an error we just want to return the existing total correct? We may want log an error to say that the API called failed, @fahlmant whats your thoughts on logging this error since we wont do anything with it? I think it would be good to know when the call fails eg the total account number has been consistent and we want to see if the call is failing?

Reply:
This case is taken care of in ```	if err != nil {
		log.Error(err, fmt.Sprintf("Failed to get total number of AWS accounts: %s", err))
	} else {
		localmetrics.MetricTotalAWSAccounts.With(prometheus.Labels{"name": "aws-account-operator"}).Set(float64(accountTotal))
	}
```

Reply:
I don't think it is. You're setting accountTotal to the first returned value, which is `0` in this case.

Reply:
@fahlmant only if there is no error, hes returning an error with the zero